src/routes/solid-start/guides/actions-and-mutations.mdx

```diff
@@ -0,0 +1,272 @@
+---
+title: "Actions and Mutations"
+---
+
+This guide provides practical examples of using actions to mutate data in SolidStart.
+
+## Handling form submission
+
+SolidStart extends the HTML [`<form>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form) element to allow actions to be invoked with the `action` prop. You can use it to handle form submissions:
+
+1. Create an action with a unique name
+2. Pass your action to the `<form>` element in the `action` prop
+3. Make sure the `<form>` element uses the `post` method
+
+```tsx {3-9} {13} title="src/routes/index.tsx"
+import { action } from "@solidjs/router";
+
+const addPost = action(async (formData: FormData) => {
+	const title = formData.get("title");
+	await fetch("https://my-api.com/posts", {
+		method: "POST",
+		body: JSON.stringify({ title }),
+	});
+}, "addPost");
+
+export default function Page() {
+	return (
+		<form action={addPost} method="post">
+			<input name="title" />
+			<button>Add Post</button>
+		</form>
+	);
+}
+```
+
+When invoked in a form, the action automatically receives the [`FormData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData) object. You can extract the field data using the native `FormData` methods.
+
+If your action accepts typed data, you can use the `with` method to pass custom data to the action:
+
+```tsx {14} title="src/routes/index.tsx"
+import { createSignal } from "solid-js";
+import { action } from "@solidjs/router";
+
+const addPost = action(async (title: string) => {
+	await fetch("https://my-api.com/posts", {
+		method: "POST",
+		body: JSON.stringify({ title }),
+	});
+}, "addPost");
+
+export default function Page() {
+	const [title, setTitle] = createSignal("");
+	return (
+		<form action={addPost.with(title())} method="post">
+			<input value={title()} onChange={(e) => setTitle(e.target.value)} />
+			<button>Add Post</button>
+		</form>
+	);
+}
+```
+
+Note that the action takes a `string` as its parameter rather than a `FormData`.
```

Comment:
```suggestion
const addPost = action(async (title: string, formData: FormData) => {
	await fetch("https://my-api.com/posts", {
		method: "POST",
		body: JSON.stringify({ title }),
	});
}, "addPost");

export default function Page() {
	const [title, setTitle] = createSignal("");
	return (
		<form action={addPost.with(title())} method="post">
			<input value={title()} onChange={(e) => setTitle(e.target.value)} />
			<button>Add Post</button>
		</form>
	);
}
```

Note that the action now takes a `string` as its first parameter and the `FormData` as its second.
```

Reply:
I tried to not go into details of APIs in this guide. I believe that this content is better suited for the API Reference.

Reply:
I understand, but saying
> the action takes a `string` as its parameter rather than a `FormData`
just felt a little misleading, it gives off the impression that the formdata cannot be used anymore if you switch to the `.with` form.

Reply:
How about this:

> Note that the action takes a `string` as its **first** parameter rather than a `FormData`.

This doesn't imply that `FormData` is not accessible inside the action and it doesn't go too much API details.

Reply:
Do you have a practical example of using `FormData` inside an action when using `with()`? If there is a common pattern that I'm not aware of, I would update this particular example.

Reply:
I think this wording makes more sense to me now, I'm good with it.

Also [here's an example](https://www.jacobparis.com/content/remix-multiple-actions#multiple-actions-from-a-single-form) that shows one use case with Remix where we would need both arguments

Instead of the hidden input like in this example, we would pass the tweet id using `.with`, and use the formdata to determine which action was taken.

Reply:
I completely rewrote this section. Instead of focusing solely on typed data, I shifted the emphasis to passing additional data. This change better represents the API, and it implicitly suggests that the additional arguments are typed, so there's no need to state that explicitly. 

What do you think about this approach?